test/asb-helper.js

```diff
@@ -0,0 +1,57 @@
+const { ReceiveMode, ServiceBusClient } = require('@azure/service-bus')
+const config = require('../server/config/mq-config')
+
+// When calling this witin a test script, ensure there is a generous timeout
+// for the connections to complete within. `30000` should be good
+async function clearQueue (queueName) {
+  // There are three queues with potentially three different hosts and
+  // credentials, however, atm there is only the single instance
+  let sbClient
+  try {
+    const connectionString = `Endpoint=sb://${config.claimQueue.host}/;SharedAccessKeyName=${config.claimQueue.username};SharedAccessKey=${config.claimQueue.password}`
+    sbClient = ServiceBusClient.createFromConnectionString(connectionString)
+
+    const queueAddress = config[queueName].address
+    const queueClient = sbClient.createQueueClient(queueAddress)
+    const receiver = queueClient.createReceiver(ReceiveMode.peekLock)
+
+    console.log(`Setup to receive messages from '${queueAddress}'.`)
+    // there shouldn't be more than a handful of messages in any queue and they
+    // are recieved in blocks of `batchSize`, in as many `batches` as specified
+    const batches = 10
```

Comment:
Could we not change this to a `do while` loop and just receive a batch each time, exiting the loop when the volume received is lower than the batch size?  Then we can be confident that the queue is empty and not just beyond the capacity of our batch size.

Reply:
Yes, I've made the change 👍